client/src/components/brushableHistogram/utils.ts

```diff
@@ -0,0 +1,41 @@
+export function truncateText(
```

Comment:
what are your thoughts on this approach - with binary search and the `getComputedTextLength` is extracted to a helper function. We can also potentially memoize `measureText` value but not sure it is necessary here

```
export function truncateText(
  text: string,
  node: SVGTextElement,
  maxWidth: number
) {
  // Helper function to measure text length
  const measureText = (str: string) => {
    node.textContent = str;
    return node.getComputedTextLength();
  };

  // Initial check if truncation is needed
  if (measureText(text) <= maxWidth) return text;

  // Binary search for the truncation point
  let start = 0;
  let end = text.length - 1;
  let truncatedText = "";

  while (start <= end) {
    const mid = Math.floor((start + end) / 2);
    const candidate = `${text.slice(0, mid)}...`;
    const candidateLength = measureText(candidate);

    if (candidateLength <= maxWidth) {
      truncatedText = candidate;
      start = mid + 1;
    } else {
      end = mid - 1;
    }
  }

  return truncatedText;
}

```

Reply:
Great callout! Yeah I actually started with binary search and then ChatGPT suggested that using the text length average as the first guess should get us closer to the target length faster (since it should save a first few binary search iterations that are far from the target)

But also what @rainandbare said, maybe I can just just CSS truncate and text-overflow style to achieve this without hogging the JS thread 😆 

Let me try that real quick!

Thanks both!

Reply:
follow up 😭 

https://github.com/chanzuckerberg/single-cell-explorer/pull/1054#discussion_r1692037751